src/pal/src/misc/stackstring.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+#include "pal/malloc.hpp"
+#include "pal/dbgmsg.h"
+
+SET_DEFAULT_DEBUG_CHANNEL(MISC);
+
+template <SIZE_T MEMSIZE>
+class StackString
+{
+private:
+    char m_innerBuffer[MEMSIZE];
+    char *m_buffer;
+	SIZE_T m_allocation; // actual allocated size
+
+public:
+	StackString(SIZE_T size)
+	: m_allocation(0), m_buffer(NULL)
+	{
+		ReallocateBuffer(size);
+	}
+	
+	StackString()
+	: m_allocation(0), m_buffer(NULL)
+	{
```

Comment:
Do we need to initialize m_buffer to point to m_innerBuffer here? I think that not doing this can cause an AV if the Set method is called with size <=MEMSIZE.